packages/angular/src/rule-builder/rule-builder.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { I18n, ListItem } from 'carbon-components-angular';
+import {
+  filterRulesById,
+  findRulePathById,
+  generateRule,
+  generateRuleGroup,
+  insertRuleAfterPath,
+} from './utils';
+
+@Component({
+  selector: 'ai-rule-builder',
+  template: `
+    <div>
+      <ai-rule-builder-header
+        [(groupLogic)]="tree.groupLogic"
+        (addRule)="handleAddRule($event.id, $event.isGroup)"
+      ></ai-rule-builder-header>
+      <ng-container *ngFor="let rule of tree.rules; let i = index">
+        <ai-rule
+          (addRule)="handleAddRule($event.id, $event.isGroup)"
+          (removeRule)="handleRemoveRule($event)"
+          [columns]="columns"
+          [columnOperands]="columnOperands"
+          [(rule)]="tree.rules[i]"
```

Comment:
can we just use `rule`?

Reply:
No, not for double binding. Angular.
Story works fine, build fails. This is a workaround.